src/context/WalletProvider/WalletProvider.tsx

```diff
@@ -172,6 +181,74 @@ export const WalletProvider = ({ children }: { children: React.ReactNode }): JSX
     }
   }, [state.keyring])
 
+  useEffect(() => {
+    const localWalletType = getLocalWalletType()
+    const localWalletDeviceId = getLocalWalletDeviceId()
+    if (localWalletType && state.adapters) {
+      ;(async () => {
+        if (state.adapters?.has(localWalletType)) {
+          if (localWalletType === KeyManager.Native) {
+            const localWallet = await state.adapters
+              .get(localWalletType)
+              ?.pairDevice(localWalletDeviceId)
+            if (localWallet) {
+              await localWallet.initialize()
+            } else {
+              clearLocalWallet()
+            }
+          } else if (localWalletType === KeyManager.KeepKey) {
+            try {
+              const localWallet = await state.adapters.get(KeyManager.KeepKey)?.pairDevice()
+              if (localWallet) {
+                const { name, icon } = SUPPORTED_WALLETS[KeyManager.KeepKey]
+                const deviceId = await localWallet.getDeviceID()
+                // This gets the firmware version needed for some KeepKey "supportsX" functions
+                await localWallet.getFeatures()
+                // Show the label from the wallet instead of a generic name
+                const label = (await localWallet.getLabel()) || name
+
+                await localWallet.initialize()
+
+                dispatch({
+                  type: WalletActions.SET_WALLET,
+                  payload: { wallet: localWallet, name: label, icon, deviceId, meta: { label } }
+                })
+                dispatch({ type: WalletActions.SET_IS_CONNECTED, payload: true })
+              } else {
+                dispatch({ type: WalletActions.SET_LOCAL_WALLET_LOADING, payload: false })
+                clearLocalWallet()
+              }
+            } catch (e) {
+              clearLocalWallet()
+            }
+          } else {
```

Comment:
I didn't understand what you mean, could you elaborate more?

Reply:
I mean, handling all types explicitly instead of an else/default: each type should be explicilty checked, so we can later on simply add XDefi here instead of having the default catch all other cases.